nucypher/network/nodes.py

```diff
@@ -1096,11 +1078,11 @@ def validate_worker(self, registry: BaseContractRegistry = None) -> None:
         # Decentralized
         else:
 
-            # Off-chain signature verification
-            if not self._stamp_has_valid_signature_by_worker():
-                message = f"Invalid signature {self.__decentralized_identity_evidence.hex()} " \
-                          f"from worker {self.worker_address} for stamp {bytes(self.stamp).hex()} "
-                raise self.InvalidWorkerSignature(message)
+            # Try to derive the worker address if it hasn't been derived yet.
+            try:
+                worker_address = self.worker_address
```

Comment:
I know it's a bit of extra lift however, this line will read much more explicitly and better fit the emerging conventions as a function.